sd/etcd/registrar.go

```diff
@@ -0,0 +1,54 @@
+package etcd
+
+import (
+	etcd "github.com/coreos/etcd/client"
+	"github.com/go-kit/kit/log"
+)
+
+// Registrar registers service instance liveness information to etcd.
+type Registrar struct {
+	client  Client
+	service Service
+	logger  log.Logger
+}
+
+// Service holds the key, value and instance identifying data you
+// want to publish to etcd.
+type Service struct {
+	Key           string // discovery key, example: /myorganization/myplatform/
+	Value         string // service name value, example: addscv
+	Data          []byte // instance data to store for discovery, example: 10.0.2.10:80
```

Comment:
I had to take a good look at this one considering the time since implementation. Im not sure why I added this `Data` field but from everything I can tell it appears to be useless. I removed it from my test suite as well as a local client I created to actively test against a long standing etcd instance with no issues. I am removing this field unless I can otherwise quantify its reason for existence.